web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/SickLeaveInformationServiceImpl.java

```diff
@@ -60,7 +67,8 @@ private static void setEmployeeNameIfFound(SjukfallEnhet sickLeave, String emplo
         }
     }
 
-    private String getHsaEmployee(String doctorId) {
+    @Cacheable(value = EMPLOYEE_NAME_CACHE_NAME, key = "#doctorId")
+    public String getHsaEmployee(String doctorId) {
```

Comment:
If you call this from inside the class, it won't use the cache. Maybe consider moving this to HsaService instead.

Add a LOG.error in the catch-block and log the exception catched. We always want to have the original exception logged.

Reply:
Move it into HsaEmployeeService.

Reply:
Realised I used wrong implementation of HsaEmployeeService, I used the legacy version. Changed to the correct one that already had caching implemented.